CryptoPrices/Sources/MyCoin/Sources/MyCoin/ChartSectionView/PriceLineChartSection.swift

```diff
@@ -0,0 +1,108 @@
+        ChartDataEntry(x: 1641254400000, y: 46531.140860530526),
+        ChartDataEntry(x: 1641340800000, y: 45938.02427172366),
+        ChartDataEntry(x: 1641427200000, y: 43647.147508068054)
+    ]
+
+    var body: some View {
+
+        // TODO: Update to use real data in the integrate task
+        GeometryReader { geometry in
+            VStack(alignment: .center, spacing: 0.0) {
+                Text("50774.067814743")
+                    .font(Fonts.Inter.medium.textStyle(.caption))
+                    .foregroundColor(Colors.guppieGreen.swiftUIColor)
+                    .readSize { textSize in
+                        highestPriceOffsetX = calculateDataPointOffsetX(
+                            data: testData,
+                            parentViewWidth: geometry.size.width,
+                            labelWidth: textSize.width
+                        )
+                    }
+                    .offset(x: highestPriceOffsetX)
+                PriceLineChartView(entries: testData)
+                Text("45938.02427172366")
+                    .font(Fonts.Inter.medium.textStyle(.caption))
+                    .foregroundColor(Colors.carnation.swiftUIColor)
+                    .readSize { textSize in
+                        lowestPriceOffsetX = calculateDataPointOffsetX(
+                            data: testData,
+                            parentViewWidth: geometry.size.width,
+                            labelWidth: textSize.width,
+                            isHighest: false
+                        )
+                    }
+                    .offset(x: lowestPriceOffsetX)
+            }
+        }
+    }
+}
+
+private extension PriceLineChartSection {
+
+    func calculateDataPointOffsetX(
+        data: [ChartDataEntry],
+        parentViewWidth: CGFloat,
+        labelWidth: CGFloat,
+        isHighest: Bool = true
+    ) -> CGFloat {
+        var entry: ChartDataEntry?
+        if isHighest {
+            entry = data.max { $0.y < $1.y }
+        } else {
+            entry = data.min { $0.y < $1.y }
+        }
+        guard let finalEntry = entry,
+              let index = data.firstIndex(of: finalEntry) else { return 0.0 }
+        // All data points from server has the same time interval
+        let spaceInterval = parentViewWidth / CGFloat(data.count - 1)
+        let positionX = spaceInterval * CGFloat(index)
+        let halfParentViewWidth = parentViewWidth / 2.0
+        let halfLabelWidth = labelWidth / 2.0
+
+        // Calculate the price label position if it exceeds the trailing/leading of the screen or not
+        if positionX + halfLabelWidth > parentViewWidth { // Exceed the trailing
+            return halfParentViewWidth - halfLabelWidth
+        } else if positionX - halfLabelWidth < 0 { // Exceed the leading
+            return -halfParentViewWidth + halfLabelWidth
+        } else { // Within the screen region
+            return positionX - halfParentViewWidth
+        }
```

Comment:
It's great when you've taken into account the edge cases here 🙌

I just have a minor suggestion that we can add a little more extra breathing space for the cases of exceeding the screen:
```swift
if positionX + halfLabelWidth > parentViewWidth { // Exceed the trailing
  return halfParentViewWidth - halfLabelWidth - 8.0  👈 
} else if positionX - halfLabelWidth < 0 { // Exceed the leading
  return -halfParentViewWidth + halfLabelWidth + 8.0  👈 
} else {
  // ...
```

And this shows what it does (formatting data with `dollarCurrency`)

![Simulator Screen Shot - iPhone 14 - 2023-01-05 at 15 08 45](https://user-images.githubusercontent.com/17972674/210733517-db475334-576c-475a-a2a5-d32e8774423f.png)

Reply:
Having the padding is a good suggestion, will update 👌